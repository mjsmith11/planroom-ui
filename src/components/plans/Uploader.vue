<template>
    <div class='uploader'>
      <h4>Add Bidding Documents</h4>
      <div class="alert alert-danger" v-if="uploadProblem">
        <strong>Warning!</strong> One or more documents failed to upload
      </div>
      <vue-dropzone
        :awss3="awss3"
        v-on:vdropzone-s3-upload-error="s3UploadError"
        v-on:vdropzone-s3-upload-success="s3UploadSuccess"
        id=''
        :options='dzOpts'
        >
      </vue-dropzone>
    </div>
</template>
<script>
import vue2Dropzone from 'vue2-dropzone'
import 'vue2-dropzone/dist/vue2Dropzone.min.css'
import { EventBus } from '../../event-bus.js'
export default {
  components: {
    vueDropzone: vue2Dropzone
  },
  data () {
    return {
      awss3: {
        signingURL: this.getUrl,
        headers: {'Planroom-Authorization': 'Bearer ' + this.$store.getters.token},
        params: {},
        sendFileToServer: false,
        withCredentials: false
      },
      dzOpts: {
        url: '',
        maxFilesize: 1024,
        timeout: 15 * 60 * 1000 // 15 minutes
      },
      uploadProblem: false
    }
  },
  methods: {
    s3UploadError (errorMessage) {
      this.uploadProblem = true
    },
    s3UploadSuccess (s3ObjectLocation) {
      EventBus.$emit('file-uploaded')
    },
    getUrl (file) {
      return process.env.VUE_APP_API_URL + '/jobs/' + this.jobId + '/plans?filename=' + file.name
    }
  },
  props: {
    jobId: Number
  }
}
</script>
<style>
  .uploader {
    width: 100%;
    margin-top: 25px;
  }
</style>
