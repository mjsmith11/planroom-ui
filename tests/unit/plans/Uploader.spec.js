import { mount, createLocalVue } from '@vue/test-utils'
import Uploader from '@/components/plans/Uploader'
// import mockAxios from 'jest-mock-axios'
import { EventBus } from '@/event-bus.js'
import Vuex from 'vuex'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Uploader', () => {
  let cmp
  let getters = {
    token: () => 'eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjEyMzQ1Njc3ODc3fQ.6xV-z88Nvmag8i4jVwmOZjX3MhCYAgb3rqttN4ROix3EbtHLwYIG3utNVaCpCN2cS7QFAJM3CPnfiS5_s9luiA'
  }
  let store = new Vuex.Store({
    getters
  })

  beforeEach(() => {
    cmp = mount((Uploader), {
      propsData: {
        jobId: 1
      },
      store,
      localVue
    })
  })

  afterEach(() => {
    // cleaning up the mess left behind the previous test
    // mockAxios.reset()
  })

  it('has expected html structure', () => {
    expect(cmp.html()).toMatchSnapshot()
  })

  it('alerts on failed upload', () => {
    cmp.vm.s3UploadError('')
    expect(cmp.vm.uploadProblem).toBeTruthy()
    const alerts = cmp.findAll('div.alert')
    expect(alerts.length).toBe(1)
    expect(alerts.at(0).classes()).toContain('alert-danger')
    expect(alerts.at(0).html()).toContain('<strong>Warning!</strong> One or more documents failed to upload')
  })

  it('emits file-uploaded event', () => {
    let spy = jest.spyOn(EventBus, '$emit')
    cmp.vm.s3UploadSuccess('')
    expect(spy).toBeCalledWith('file-uploaded')
  })

  it('builds signing URL', () => {
    let file = {
      name: 'myfile.pdf'
    }
    let result = cmp.vm.getUrl(file)
    expect(result).toBe('http://test-api.com/jobs/1/plans?filename=myfile.pdf')
  })
})
