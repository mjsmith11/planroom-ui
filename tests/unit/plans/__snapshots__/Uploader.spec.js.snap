// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader has expected html structure 1`] = `
<div class="uploader">
  <h4>Add Bidding Documents</h4>
  <!---->
  <div id="" class="vue-dropzone dropzone dz-clickable">
    <!---->
    <div class="dz-default dz-message"><span>Drop files here to upload</span></div>
  </div>
</div>
`;
